third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -765,6 +739,42 @@ impl<'a> FunctionGenerator<'a> {
         }
     }
 
+    /// Ensures that all `temps` which are on the stack and used after this program
+    /// point are saved to locals. This flushes the stack as deep as needed for this.
+    fn save_used_after(&mut self, ctx: &BytecodeContext, temps: &[TempIndex]) {
+        let mut stack_to_flush = self.stack.len();
+        for temp in temps {
+            if let Some(pos) = self.stack.iter().position(|t| t == temp) {
+                if ctx.is_alive_after(*temp) {
+                    // Determine new lowest point to which we need to flush
+                    stack_to_flush = std::cmp::min(stack_to_flush, pos);
+                }
+            }
+        }
+        // Notice that we flush the stack _before_ the next processed instruction, therefore
+        // we use the before version of the below function.
+        self.abstract_flush_stack_before(ctx, stack_to_flush)
+    }
+
+    /// Determines the maximal prefix of `temps` which are already on the stack, and
+    /// returns the temps which are not and need to be pushed.
+    fn analyze_stack<'t>(
+        &mut self,
+        _ctx: &BytecodeContext,
```

Comment:
Why do we need this argument in this function?

Reply:
+1

Reply:
Done